dusty/systems/nfs/client.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import absolute_import
+
+import logging
+from subprocess import CalledProcessError
+import time
+
+from ...log import log_to_client
+from ...subprocess import call_demoted, check_call_demoted, check_output_demoted
+from ...compiler.spec_assembler import get_all_repos
+
+def mount_active_repos():
+    remount_repos(get_all_repos(active_only=True, include_specs_repo=False))
+
+def remount_repos(repos):
+    _start_nfs_client()
+    for i, repo in enumerate(repos):
+        wait_for_server = not i
+        _unmount_repo(repo)
+        _mount_repo(repo, wait_for_server=wait_for_server)
+
+def _start_nfs_client():
+    check_call_demoted(['boot2docker', 'ssh', 'sudo /usr/local/etc/init.d/nfs-client start'])
+
+def _unmount_repo(repo):
+    call_demoted(['boot2docker', 'ssh', 'sudo umount -l {}'.format(repo.vm_path)])
+
+def _mount_repo(repo, wait_for_server=False):
+    check_call_demoted(['boot2docker', 'ssh', 'sudo mkdir -p {}'.format(repo.vm_path)])
+    if wait_for_server:
+        for i in range(0,10):
+            try:
+                _run_mount_command(repo)
+                return
+            except CalledProcessError as e:
+                if 'Connection refused' in e.output:
+                    logging.info('Failed to mount repo; waiting for nfsd to restart')
+                    time.sleep(1)
+                else:
+                    raise e
+        log_to_client('Failed to mount repo {}'.format(repo.short_name))
+        raise RuntimeError('Unable to mount repo with NFS')
+    else:
+        _run_mount_command(repo)
+
+def _run_mount_command(repo):
+    return check_output_demoted(['boot2docker', 'ssh', 'sudo mount {}'.format(_nfs_mount_args_string(repo))])
```

Comment:
Ah, ok. This is brittle as-is, the caller's dependency on this is implicit. At a minimum, you should add a comment explaining why it needs to stay `output`, and if possible you should refactor it to make the dependency explicit.